### DISCUSSION TITLE
Allow to sync sync_dir to a specific remote folder

### DISCUSSION THREAD

**Discussion by sromeroi** [abraunegg/onedrive#1675] [Q&A] [answered]
I just installed the OneDrive client in Linux Mint from OpenSuSE build service and couldn't find an option to synchronize "~/OneDrive" to a folder named "DesktopSync" in my OneDrive storage.

My OneDrive storage keeps right now several "backup" folders from a NAS and I don't want to synchronize them, just a simple sync between my laptop (~/OneDrive folder) and a remote folder in OneDrive (/DesktopSync):

(local) ~/OneDrive <---> (remote) /DesktopSync

(which will keep a file sync between my OneDrive folder and the DesktopSync folder in the remote Service)

Instead of:

(local) ~/OneDrive <---> (remote) /

(which will sync to my local computer all the folders in the root of the OneDrive service, including my NAS backup, Photo uploads, and some folders I don't want to be synced).

I also checked if skip_dir allowed to specify remote folders but it looks it's meant for local folders (I would skipped all remote folders but for "/DesktopSync").

Is this possible with the current client?

Thanks.

**Comment by abraunegg** [comment, marked as answer]
This is not directly possible with the client because of limitations of the OneDrive API.

This is achievable however, but you need to look at how you structure your local folders.

1. On your local filesystem, create a new path, for example: `/opt/OneDrive`. Configure the application 'sync_dir' to point to this location.
2. Within your home directory, symbolicly link `~/OneDrive` -> `/opt/OneDrive/DesktopSync`

This will give you:
* Your `~/OneDrive` will look / feel as per normal
* The data will be stored online under `/DesktopSync`

This is the *only* way to achieve this sort of syncing to a specific folder.

**Reply by sromeroi** [reply]
Sorry for opening it as a Feature Request...

Yes, your solution would avoid having ~/OneDrive/DesktopSync/ pointing to (remote)/DesktopSync.

But, this also means that I will sync/download to my laptop (under /opt/OneDrive) all my remote backup folders that I don't want to download such as (example):

(Remote)/NAS_Backup  <---> /opt/OneDrive/NAS_backup (almost 400GB)
(Remote)/PhotoUploads  <---> /opt/OneDrive/PhotoUploads (typically 100GB)
(etc)

Is this correct? 

Or... at that point I could use skip_dir to "ignore" "NAS_Backup", "PhotoUploads", etc?

(I can't sync 700GB of remote data in this laptop with 256GB of Hard Disk)...

Thanks.

**Reply by abraunegg** [reply]
@sromeroi 
What you would need to do is then configure the client to only sync the desired folder - see:
* sync_list
* --single-directory

**Reply by sromeroi** [reply]
That's perfect, let me check it right now :)
